protocols/mdns/tests/smoke.rs

```diff
@@ -78,24 +80,121 @@ async fn run_test(config: MdnsConfig) -> Result<(), Box<dyn Error>> {
-    run_test(MdnsConfig::default()).await
+    run_discovery_test(config).await
 }
+
+async fn run_peer_expiration_test(config: MdnsConfig) -> Result<(), Box<dyn Error>> {
+    let mut a = create_swarm(config.clone()).await?;
+    let mut b = create_swarm(config).await?;
+
+    loop {
+        futures::select! {
+            ev = a.select_next_some() => match ev {
+                SwarmEvent::Behaviour(MdnsEvent::Expired(peers)) => {
+                    for (peer, _addr) in peers {
+                        if peer == *b.local_peer_id() {
+                            return Ok(());
+                        }
+                    }
+                }
+                _ => {}
+            },
+            ev = b.select_next_some() => match ev {
+                SwarmEvent::Behaviour(MdnsEvent::Expired(peers)) => {
+                    for (peer, _addr) in peers {
+                        if peer == *a.local_peer_id() {
+                            return Ok(());
+                        }
+                    }
+                }
+                _ => {}
+            }
+
+        }
+    }
+}
+
+#[async_std::test]
+async fn test_expired_async_std_ipv4() -> Result<(), Box<dyn Error>> {
+    let config = MdnsConfig {
+        ttl: Duration::from_millis(500),
+        query_interval: Duration::from_secs(1),
+        ..Default::default()
+    };
+
+    run_timebound_test(
+        run_peer_expiration_test(config),
+        Duration::from_secs(6)
+    ).await
+}
+
+#[async_std::test]
+async fn test_expired_async_std_ipv6() -> Result<(), Box<dyn Error>> {
+    let config = MdnsConfig {
+        ttl: Duration::from_millis(500),
+        query_interval: Duration::from_secs(1),
+        multicast_addr: *IPV6_MDNS_MULTICAST_ADDRESS,
+    };
+
+    run_timebound_test(
+        run_peer_expiration_test(config),
+        Duration::from_secs(6)
+    ).await
+}
+
+#[tokio::test]
+async fn test_expired_tokio_ipv4() -> Result<(), Box<dyn Error>> {
+    let config = MdnsConfig {
+        ttl: Duration::from_millis(500),
+        query_interval: Duration::from_secs(1),
+        ..Default::default()
+    };
+
+    run_timebound_test(
+        run_peer_expiration_test(config),
+        Duration::from_secs(6)
+    ).await
+}
+
+#[tokio::test]
+async fn test_expired_tokio_ipv6() -> Result<(), Box<dyn Error>> {
+    let config = MdnsConfig {
+        ttl: Duration::from_millis(500),
+        query_interval: Duration::from_secs(1),
+        multicast_addr: *IPV6_MDNS_MULTICAST_ADDRESS,
+    };
+
+    run_timebound_test(
+        run_peer_expiration_test(config),
+        Duration::from_secs(6)
+    ).await
+}
+
+async fn run_timebound_test(fut: impl Future<Output=Result<(), Box<dyn Error>>>, bound: Duration) -> Result<(), Box<dyn Error>> {
+    let result = async_std::future::timeout(bound, fut)
+        .await;
+
+    match result {
+        Ok(res) => res,
+        Err(err) => Err::<(), Box<dyn Error>>(Box::new(err))
+    }
```

Comment:
```suggestion
    async_std::future::timeout(bound, fut)
        .await
        .map(|_| ())
        .map_err(|e| Box::new(e) as Box<dyn Error>)
```

Would be a bit simpler, no?

Reply:
Indeed, In fact, I noticed I always used `async_std` and skipped `tokio`. this is also fixed.